### PR TITLE
Feature/offline cdr logging

### PR DESCRIPTION
**Pull request by nickvsnetworking**

### CODE REVIEW

File: src/sgwu/gtp-path.c [hunk 1]
```diff
@@ -192,17 +207,44 @@ static void _gtpv1_u_recv_cb(short when, ogs_socket_t fd, void *data)
...
+            sess = SGWU_SESS(pdr->sess);
+            ogs_assert(sess);
+
+            ogs_pfcp_far_t *far = NULL;
+            far = pdr->far;
+            ogs_assert(far);
+
+            /* Check if FAR is Downlink */
+            if (far->dst_if == OGS_PFCP_INTERFACE_ACCESS)
+                ogs_info("far is downlink");
```
Comment (nickvsnetworking) [inline src/sgwu/gtp-path.c]: Should this set ```is_ul = false``` ?
It's referenced on :242

Comment (nickvsnetworking) [reply src/sgwu/gtp-path.c]: Disregard - I can see it's set on :222

File: src/sgwu/context.c [hunk 2]
```diff
@@ -250,3 +254,220 @@ sgwu_sess_t *sgwu_sess_add_by_message(ogs_pfcp_message_t *message)
...
+
+void sgwu_sess_urr_acc_add(sgwu_sess_t *sess, ogs_pfcp_urr_t *urr, size_t size, bool is_uplink)
+{
+    sgwu_sess_urr_acc_t *urr_acc = &sess->urr_acc[urr->id];
+    uint64_t vol;
+
+    /* Increment total & ul octets + pkts */
+    urr_acc->total_octets += size;
+    urr_acc->total_pkts++;
+    if (is_uplink) {
```
Comment (nickvsnetworking) [inline src/sgwu/context.c:280]: So for whatever reason this is not being hit...

File: src/sgwu/context.c [hunk 3]
```diff
@@ -250,3 +254,220 @@ sgwu_sess_t *sgwu_sess_add_by_message(ogs_pfcp_message_t *message)
 
     return sess;
 }
+
+void sgwu_sess_urr_acc_add(sgwu_sess_t *sess, ogs_pfcp_urr_t *urr, size_t size, bool is_uplink)
```
Comment (nickvsnetworking) [inline src/sgwu/context.c:272]: It's called is_uplink here but is_ul elsewhere, would this have anything to do with it?

File: configs/open5gs/sgwc.yaml.in [hunk 4]
```diff
@@ -82,6 +82,11 @@ sgwc:
       - addr: 127.0.0.3
     pfcp:
       - addr: 127.0.0.3
+    cdr:
+      enabled: True
+      file_capture_period_sec: 1800
+      reporting_period_sec: 10
+      sgw_name: SGW-01
```
Comment (nickvsnetworking) [inline configs/open5gs/sgwc.yaml.in:90]: Do we need to add `log_dir` parameter here? @Ayrdim

Comment (Ayrdim) [reply configs/open5gs/sgwc.yaml.in:90]: Oops forgot this .in file example, yeah its been added

**Pull request merged**